### DISCUSSION TITLE
using yield_per with orm

### DISCUSSION THREAD

**Discussion by ngodber** [sqlalchemy/sqlalchemy#6579] [Usage Questions] [answered]
Hi,

I have a quite large sqlite table I'm iterating over. I would like to make use of the yield_per functionality to retrieve chunks of rows for processing. The difficulty I'm having is that I cannot find any way to get the partition of rows to return as objects. As a result I'm having to resort constructing SELECT statements using either WHERE id ranges or LIMIT and OFFSET then executing the statement 
eg `results = database.execute(sqlalchemy.select(base.Transect).limit(50)).scalars()`

Am I missing something or is ORM not supported for the yield_per functionality? Not the end of the world of course, just a bit clunky.

**Comment by zzzeek** [comment]
yield_per is an ORM option and is intended to be used with ORM objects.

as for why you mention "cannot find any way", I would need more detail on that, you should be able to use the method as illustrated at https://docs.sqlalchemy.org/en/14/orm/queryguide.html#yield-per .

**Comment by ngodber** [comment, marked as answer]
Ah I think I see. Yes I had found that documentation but you can see the point of confusion from below, .scalars() etc yield a list of objects where as partition yields as list of rows of which the zeroth object is the actual object itself. So the behavior is subtly different which led to my confusion. eg.

![image](https://user-images.githubusercontent.com/11834846/120874188-dc0f3200-c5e8-11eb-888f-97981f9fd304.png)

which yields.

![image](https://user-images.githubusercontent.com/11834846/120874195-e3ced680-c5e8-11eb-9353-4d7b9ca7561c.png)

Adding row[0] gives the desired outcome:

![image](https://user-images.githubusercontent.com/11834846/120874326-7f604700-c5e9-11eb-9d5e-cfe4e8b7bb9b.png)


Extensive googling didn't show this. But now this exists so hopefully others in a similar position find this answer! Cheers :)

**Reply by CaselIT** [reply]
I believe you can do `scalars().partition(10)` ( or the other way around) to get a list of objects instead of Rows
